src/scripts/testbot/coverage_targets.py

```
@@ -0,0 +1,252 @@
+    owner: str = "NVIDIA",
+    repo: str = "OSMO",
+    branch: str = "main",
+) -> dict:
+    """Fetch the coverage report JSON from Codecov API."""
+    url = f"{CODECOV_API_BASE}/github/{owner}/repos/{repo}/report/?branch={branch}"
+    request = urllib.request.Request(
+        url,
+        headers={"Authorization": f"token {token}"},
+    )
+    try:
+        with urllib.request.urlopen(request, timeout=60) as response:
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)